tests/integration-tests/src/commonTest/kotlin/test/BodyExtensionsTest.kt

```diff
@@ -71,4 +71,60 @@ class BodyExtensionsTest {
     apolloClient.close()
     mockServer.close()
   }
+
+  @Suppress("UNCHECKED_CAST")
+  @Test
+  fun enhancedClientAwarenessExtensionsIncludedByDefault() = runTest {
+    val mockServer = MockServer()
+
+    val apolloClient = ApolloClient.Builder()
+        .networkTransport(
+            HttpNetworkTransport.Builder()
+                .httpRequestComposer(DefaultHttpRequestComposer(mockServer.url()))
+                .build()
+        )
+        .build()
+
+    mockServer.enqueueError(statusCode = 500)
+    apolloClient.query(LaunchDetailsQuery("42")).execute()
+
+    val request = mockServer.awaitRequest()
+
+    @Suppress("UNCHECKED_CAST")
+    val asMap = Buffer().write(request.body).jsonReader().readAny() as Map<String, Any>
+    val expected: Map<String, Any> = mapOf("name" to "apollo-kotlin", "version" to com.apollographql.apollo.api.apolloApiVersion)
+
+    assertEquals(expected, (asMap["extensions"] as Map<String, Any>).get("clientLibrary"))
+
+    apolloClient.close()
+    mockServer.close()
+  }
+
+  @Suppress("UNCHECKED_CAST")
+  @Test
+  fun enhancedClientAwarenessExtensionsExcludedWhenDisabled() = runTest {
+    val mockServer = MockServer()
+
+    val apolloClient = ApolloClient.Builder()
+        .sendEnhancedClientAwareness(false)
```

Comment:
@martinbonnin / @BoD  - this does not appear to be working as intended but I'm unsure if I'm just not constructing things correctly.

Reply:
I think it's because the value needs to be passed over from the client to the request inside `ApolloClient.apolloResponses()`